Source/Glass.Mapper.Sc.Mvc/Web/Mvc/IMvcContext.cs

```diff
@@ -170,8 +171,11 @@ public virtual Item DataSourceItem
         {
             get
             {
-                var dataSource = RenderingContext.CurrentOrNull.Rendering.DataSource;
-               
+                // Cannot use RenderingContext.CurrentOrNull.Rendering.DataSource as AB test items 
+                // do not reuturn Guid e.g. {0A275E4A-98DF-4CB3-8A7E-948F53010AE3} instead returning a
+                // string e.g. sitecore://{0A275E4A-98DF-4CB3-8A7E-948F53010AE3}?lang=en&ver=0
+                var dataSource = RenderingContext.CurrentOrNull.Rendering.Items.FirstOrDefault().ID;

```

Comment:
If you're using FirstOrDefault() then I'd suggest null checks too, e.g.

```
var dataSource = RenderingContext.CurrentOrNull?.Rendering?.Items?.FirstOrDefault()?.ID;
if (dataSource != null) 
{

var item = MvcSettings.ItemLocator.GetItem(dataSource);
                return item;

}
```

Reply:
Null check added, thanks @markgibbons25